### PR TITLE
[chore] migrate remaining deprecated chrome storage API calls to ConfigStorage JSON implementation.

### PR DESCRIPTION
**Pull request by haslinghuis**
We need to migrate deprecated API's.

This PR is removing `chrome.storage API` and uses the `ConfigStorage.js` we had in place already instead.

Deprecated API | Suggested API | Link
-- | -- | --
chrome.storage | Cache API | https://web.dev/storage-for-the-web/
chrome.fileSystem| Native FileSystem API | https://web.dev/file-system-access/
chrome.serial | Web Serial API | https://web.dev/serial/
chrome.sockets.tcp | WebSockets | https://web.dev/websocketstream/
chrome.usb | Web USB API | https://web.dev/usb/

### CODE REVIEW

File: src/js/FirmwareCache.js
Comment (haslinghuis) [reply src/js/FirmwareCache.js:88]: Okay tried the remaining but doesn't start configurator

Screenshot from 2021-10-29 20-43-59

[diff.txt](https://github.com/betaflight/betaflight-configurator/files/7444311/diff.txt)

**Comment by chmelevskij** [comment]
So to fix the callback issue we need to refactor the `get` as well https://github.com/betaflight/betaflight-configurator/blob/78287a182c9c4aac37f061c65332ce9bd2077770/src/js/ConfigStorage.js#L7-L32

```
    get: function(key) {
        if (Array.isArray(key)) {
            let obj = {};
            key.forEach(function (element) {
                try {
                    obj = {...obj, ...JSON.parse(window.localStorage.getItem(element))};
                } catch (e) {
                    // is okay
                }
            });
            return obj;
        } else {
            const keyValue = window.localStorage.getItem(key);
            if (keyValue) {
                let obj = {};
                try {
                    obj = JSON.parse(keyValue);
                } catch (e) {
                    // It's fine if we fail that parse
                }
                return obj;
            } else {
                return {};
            }
        }
    },
```

Then in the actual code which was using the callback, return value of `get` needs to be assigned to a variable and later on used in the callback body logic.

**Comment by haslinghuis** [comment]
@chmelevskij Thanks. I tried. but still the same. [diff.txt](https://github.com/betaflight/betaflight-configurator/files/7457061/diff.txt)

File: src/js/FirmwareCache.js
Comment (haslinghuis) [reply src/js/FirmwareCache.js]: Done.

**Comment by haslinghuis** [comment]
Squashed and rebased

**Review by limonspb — APPROVED** [review]

**Comment by limonspb** [comment]
Proposal (discussed with @haslinghuis )
leave it for 10.9 or 10.8.1

**Review by limonspb — DISMISSED** [review]

**Comment by chmelevskij** [comment]
Looks good, happy to see all of those unnecessary callbacks go 🙂

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
Rebased

**Review by chmelevskij — DISMISSED** [review]

**Comment by haslinghuis** [comment]
Rebased

**Comment by haslinghuis** [comment]
Squashed and rebased

**Review by chmelevskij — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Pull request merged**
